cloudflare/data_source_access_identity_provider.go

```diff
@@ -0,0 +1,82 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceCloudflareAccessIdentityProvider() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceCloudflareAccessIdentityProviderRead,
+
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ExactlyOneOf: []string{"zone_id", "account_id"},
+			},
+			"zone_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ExactlyOneOf: []string{"zone_id", "account_id"},
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
```

Comment:
```suggestion
```

`id` isn't an exportable field as it's an internal identifer by Terraform. `id` is added to all resources for free.